Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -0,0 +1,357 @@
+/-
+Copyright (c) 2023 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.Topology.Bases
+import Mathlib.Order.Filter.CountableInter
+import Mathlib.Topology.Compactness.Compact
+
+/-!
+# Lindelöf sets and Lindelöf spaces
+
+## Main definitions
+
+We define the following properties for sets in a topological space:
+
+* `IsLindelof s`: every nontrivial filter on `s` with the countable intersection property has
+  a clusterpoint.
+* `LindelofSpace X`: `X` is Lindelöf iff every open cover has a countable subcover.
+* `NonLindelofSpace`: a space that is not a Lindëlof space.
+
+## Main results
+
+* `isLindelof_iff_countable_subcover`: A set is Lindelöf iff every open cover has a
+  countable subcover.
+-/
+open Set Filter Topology TopologicalSpace
+
+
+universe u v
+
+variable {X : Type u} {Y : Type v} {ι : Type*}
+
+variable [TopologicalSpace X] [TopologicalSpace Y] {s t : Set X}
+
+section Lindelof
+
+/-- A set `s` is Lindelöf if for every nontrivial filter `f` with the countable intersection
+  property that contains `s`, there exists `a ∈ s` such that every set of `f`
+  meets every neighborhood of `a`. -/
```

Comment:
IMHO, "has a cluster point in `s`" is good enough. Also, please reference the filter-free definition `isLindelof_iff_countable_subcover` here.